cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -322,39 +322,48 @@ def __init__(self, app,
         logger.info("constructed %s:%s for data_server_call_back=%s", self.__class__, self, data_server_call_back)
 
     @staticmethod
-    def inspect_state(app):
+    def free_up_space(app):
         token = request.args.get('token', None)
-        if token is None:
-            # TODO what about using the RequestNotAuthorized exception?
-            # raise RequestNotAuthorized("A token must be provided.")
-            return make_response('A token must be provided.'), 403
-        try:
-            secret_key = app.config.get('conf').secret_key
-            decoded_token = tokenHelper.get_decoded_token(token, secret_key)
-            logger.info("==> token %s", decoded_token)
-        except jwt.exceptions.ExpiredSignatureError:
-            # raise RequestNotAuthorized("The token provided is expired.")
-            return make_response('The token provided is expired.'), 403
-        except jwt.exceptions.InvalidTokenError:
-            # raise RequestNotAuthorized("The token provided is not valid.")
-            return make_response('The token provided is not valid.'), 403
 
-        recent_days = request.args.get('recent_days', 3, type=float)
-        job_id = request.args.get('job_id', None)
+        app_config = app.config.get('conf')
+        secret_key = app_config.secret_key
 
-        roles = tokenHelper.get_token_roles(decoded_token)
+        output, output_code = tokenHelper.validate_token_from_request(token=token, secret_key=secret_key,
+                                                                      required_roles=['job manager'],
+                                                                      action="free_up space on the server")
 
-        required_roles = ['job manager']
-        # no need to have both, one of those two is sufficient
-        if not any(item in roles for item in required_roles):
-            lacking_roles = ", ".join(sorted(list(set(required_roles) - set(roles))))
-            message = (
-                f"Unfortunately, your privileges are not sufficient for this type of request.\n"
-                f"Your privilege roles include {roles}, but the following roles are missing: {lacking_roles}."
-            )
-            # raise RequestNotAuthorized(message=message)
-            return make_response(message), 403
+        if output_code is not None:
+            return make_response(output, output_code)
+
+        numb_folders_to_delete = int(request.args.get('folder_to_delete', 5))
+
+        list_scratch_dir = sorted(glob.glob("scratch_sid_*_jid_*"), key = os.path.getmtime)
+
+        list_scratch_dir_to_delete = list_scratch_dir[0:numb_folders_to_delete] if len(list_scratch_dir) >= 5 else list_scratch_dir
+
+        for d in list_scratch_dir_to_delete:
+            shutil.rmtree(d)
+
+        result_scratch_dir_deletion = f"Removed {len(list_scratch_dir_to_delete)} scratch directories"
```

Comment:
could you make it json with some info: pre-clean space; after-clean space, number of scratch removed and left, time spent cleaning, etc

Reply:
more information to be logged?

Reply:
>  pre-clean space; after-clean space,

BTW do you have some suggestions regarding retrieving this information? Have you already done it somewhere else?

Reply:
os.statvfs?

Reply:
Thanks, I will try it, for the moment I used `shutil.disk_usage`, it seems to be ok

Reply:
It does not matter, there are many ways, shutil is a wrapper around lower level functions.